src/Orleans/Lifecycle/LifecycleObservable.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Orleans.Runtime;
+
+namespace Orleans
+{
+    public class LifecycleObservable<TStage> : ILifecycleObservable<TStage>, ILifecycleObserver
```

Comment:
![image](https://user-images.githubusercontent.com/203839/29052031-635e65b2-7c2a-11e7-80d2-3fbb60cb31fe.png)
Ah, I thought it might work for enums, since they're convertible to ints.

Oh well.